staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -718,14 +727,16 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 			}
 		}
 
-		var scaleSpec *apiextensions.CustomResourceSubresourceScale
+		var scaleSpec = &apiextensionsinternal.CustomResourceSubresourceScale{}
 		if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceSubresources) && subresources != nil && subresources.Scale != nil {
 			equivalentResourceRegistry.RegisterKindFor(resource, "scale", autoscalingv1.SchemeGroupVersion.WithKind("Scale"))
 
-			scaleSpec = subresources.Scale
+			if err := apiextensions.Convert_v1_CustomResourceSubresourceScale_To_apiextensions_CustomResourceSubresourceScale(subresources.Scale, scaleSpec, nil); err != nil {
```

Comment:
highlighting the explicit conversion: we keep construct scale sub-resource's validator from internal types

Reply:
@sttts PTAL